indico/modules/events/editing/client/js/management/ManageService.jsx

```diff
@@ -0,0 +1,228 @@
+      {showConnect && (
+        <button type="button" className="i-button icon-puzzle" onClick={() => setConnectOpen(true)}>
+          <Translate>Connect</Translate>
+        </button>
+      )}
+      {loading && <Loader active inline />}
+      <Confirm
+        size="tiny"
+        open={disconnectOpen}
+        onCancel={() => setDisconnectOpen(false)}
+        onConfirm={disconnect}
+        confirmButton={<Button content={Translate.string('Disconnect')} negative />}
+        header={Translate.string('Disconnect editing workflow service')}
+        content={
+          <Modal.Content>
+            {forceDisconnect ? (
+              <Translate>
+                Disconnecting the editing workflow service while it is unavailable may leave your
+                editing process in an inconsistent state. Do not perform this action if you plan to
+                continue using the workflow service later.{' '}
+                <Param name="strong" wrapper={<strong />}>
+                  The current error with the workflow service is most likely temporary. Attempting
+                  to disconnect and reconnect will most likely not fix it.
+                </Param>
+              </Translate>
+            ) : (
+              <Translate>
+                Do you really want to disconnect from the editing workflow service? If your editing
+                process has already started (e.g. people submitted content to be edited),
+                disconnecting is most likely not a good idea and should only be done if you do not
+                plan to enable the service again for this event.
+              </Translate>
+            )}
+          </Modal.Content>
+        }
+      />
+      <Modal size="small" open={connectOpen}>
+        <FinalForm onSubmit={connect} subscription={{}}>
+          {fprops => (
+            <>
+              <Modal.Header content={Translate.string('Connect to editing workflow service')} />
+              <Modal.Content>
+                <Form id="connect-service-form" onSubmit={fprops.handleSubmit}>
+                  <FinalInput
+                    name="url"
+                    label={Translate.string('Service URL')}
+                    description={
+                      <Translate>
+                        Provide the URL for the service implementing your custom editing workflow.
+                      </Translate>
+                    }
+                    autoFocus
+                    required
+                    validate={async value => {
+                      setServiceURLInfo(null);
+
+                      const validationError = v.required(value) || v.url(value);
+                      if (validationError) {
+                        return validationError;
+                      }
+
+                      let resp;
+                      try {
+                        // this does send a request on each keypress, but in this particular case
```

Comment:
I agree, and this is probably the simplest approach.

But an idea to consider: What about moving this to a function and adding a small delay to check the URL?
The validation would be pending until you stop writing for, say, 500 ms. Each key press within this range would cancel the previous promise (with a spinner loading).

Reply:
That's indeed an option. To be honest, I just didn't want to implement an async debounce function ;)

Reply:
Added some debouncing. Not super pretty but I don't think async debounce can be made much prettier...

Reply:
Looks good! I would suggest a Spinner to show the loading state (setLoading(true) when starting and setLoading(false) right after), but apart from that, 👍.

Reply:
not sure if that would look good - in general i like the idea, but the question is always where to put those things... more often than not when trying to put spinners somewhere they looked weird

Reply:
My idea was this: https://react.semantic-ui.com/elements/input/#states-loading

Reply:
oh nice, i forgot that this existed. looks good indeed, and can most likely be done in a generic way directly in the FormFieldAdapter!

Reply:
![image](https://user-images.githubusercontent.com/179599/79913102-5b609700-8423-11ea-8d58-565dbd0df6c7.png)